src/txn/txn.c

```diff
@@ -1029,6 +1060,8 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_STAT_CONN_INCRV(session, txn_prepared_updates_resolved,
 	    resolved_update_count);
 
+	__txn_commit_free_cursors(session, cursors);
```

Comment:
Then here you'd need to either only call this `if (cursors != NULL)` (which will be the rare case) or have that function return immediately `if (cursors == NULL)`. The former avoids the function call overhead and would be my preference but a function call will not impact performance.

Reply:
6be3417

The function should be safe to call even on `NULL` but I've added a check anyway.